File: 04-find-second-largest-num/04-find-second-largest-num.py
def findSecond(nums):
    max_num = max(nums[0], nums[1])
    second_max_num = min(nums[0], nums[1])
    for i in range(2, len(nums)):
        if nums[i] > max_num:
            max_num, second_max_num = nums[i], max_num
        elif nums[i] < max_num and nums[i] > second_max_num:
            second_max_num = nums[i]
    return second_max_num

File: 04-find-second-largest-num/04-find-second-largest-num.js
function findSecond(nums) {
  let maxNum = Math.max(nums[0], nums[1]);
  let secondMaxNum = Math.min(nums[0], nums[1]);
  for (let i = 2; i < nums.length; i++) {
    if (nums[i] > maxNum) {
      [maxNum, secondMaxNum] = [nums[i], maxNum];
    } else if (nums[i] < maxNum && nums[i] > secondMaxNum) {
      secondMaxNum = nums[i];
    }
  }
  return secondMaxNum;
}
